### PR TITLE
encryption: add cli command `cockroach gen encryption-key`

### PR DESCRIPTION
**Pull request by ghost**
Added cli command `cockroach gen encryption-key` to generate store keys for
encryption at rest.

Encryption CLI: #19783.
Release note: None

### CODE REVIEW

File: pkg/cli/gen.go [hunk 1]
```diff
@@ -126,6 +127,57 @@ func runGenAutocompleteCmd(cmd *cobra.Command, args []string) error {
...
+
+		// 32 bytes are reserved for key ID.
+		keySize := encryptionKeySize + 32
+		b := make([]byte, keySize)
+		if n, err := rand.Read(b); err != nil || n != keySize {
+			return fmt.Errorf("failed to create key with size %d", encryptionKeySize)
+		}
+
+		// Create the file that stores the key.
+		keyFile, err := os.Create(encryptionKeyPath)
```
Comment (mberhault) [inline pkg/cli/gen.go]: We should create this file with mode `0600` (owner read/write)

**Comment by windchan7** [comment]
@mberhault Updated. But I'm a bit confused why you said this: `We should ask for the AES size, not the file size, otherwise there's little point in having this utility.`

**Comment by windchan7** [comment]
Thanks for the clarification!

**Comment by windchan7** [comment]
Done!

**Review by mberhault — APPROVED** [review]
LGTM

**Pull request merged**